fmperf/loadgen/generate-input.py

```diff
@@ -99,7 +99,10 @@ def generate_vllm_request(config, url):
 
     expected = []
     for r in get_streaming_response(response):
-        expected.append(r)
+
+        # ignore 'empty' tokens at the beginning of the output (can occur because of chunked prefills)
+        if not ((len(expected) == 0) and (r['text'] == '')):
+            expected.append(r)
```

Comment:
Are we sure that this can only happen for the first token? Is it possible that a prefill can be split across >2 iterations?

Reply:
Yes, a prefill can be split across more than 2 iterations. That happens a lot: with `max_num_batched_tokens` being 512 in the default case, a prompt of length above 1400 tokens (a value typically seen when using the fmperf option ` --from-model)` will be split in at least 3 chunks.

Processing all these chunks, except for the last one will generate an "empty token". The above lines are part of the following loop:
```
expected = []
    for r in get_streaming_response(response):

        # ignore 'empty' tokens at the beginning of the output (can occur because of chunked prefills)
        if not ((len(expected) == 0) and (r['text'] == '')):
            expected.append(r)
```
            
Consequently, as long as an "empty token" is discovered at the beginning of a sequence, it is simply ignored when generating the expected output. There is no limit on the number of iterations over which a prefill operation is split: all related "empty tokens" will be ignored becaused of the above if-statement.

Reply:
BTW. I explicitly checked for "empty tokens" only at the beginning of a sequence because we know that these exists because of the chunked prefill operation. If "empty tokens" would occur at other positions, then these are not filtered out, and any issues resulting from this, should be investigated to understand what is causing this to happen.

Reply:
Thanks @jvlunteren  - I actually misunderstood the original code but I now understand how it was working. I don't know whether "empty tokens" is something that can happen anywhere other than the beginning but I think we would also want to filter those cases out too (since it would also cause incorrect calculation of the number of tokens).